### PR TITLE
Xcode upgrade to 9.2, cordova-ios to 4.5.4 and cordova to 7.0

### PR DESCRIPTION
**Pull request by shivanibhanwal**
1. Dev Environment used - cordova -> 7.0.0,  Xcode  -> 9.4.1
2. (Circle.yml) xcode version set to 9.2
3. (Rakefile) Added Provisioning profile variables to the ENV
4. (Config.xml) - xcode 9.4.1 with ios 11.4, cordova 7.0  option was crashing, because iOS11 onwards it is must to add description messages in info.plist. Also added  Info.plist for the microphone.
5.  Removed Watchmen as it was failing on circleci, it is only needed to run test cases, incase of Ember if watchmen is not installed then test falls back to node.
6.  Downgraded ember-cli version to ~2.11 as circleCI had error -> Found incompatible module

### CODE REVIEW

File: cordova/config.xml [hunk 1]
```diff
@@ -1,88 +1,100 @@
...
-  <plugin name="cordova-plugin-actionsheet" spec="https://github.com/EddyVerbruggen/cordova-plugin-actionsheet.git"/>
-  <plugin name="cordova-plugin-camera" spec="~2.1.0"/>
-  <plugin name="cordova-plugin-splashscreen" spec="~3.1.0"/>
-  <plugin name="phonegap-plugin-barcodescanner" spec="https://github.com/wildabeast/BarcodeScanner.git"/>
-  <plugin name="com.phonegap.plugins.twilioclient" spec="https://github.com/crossroads/twilio_client_phonegap.git"/>
-  <plugin name="cordova-plugin-ios-no-export-compliance" spec="https://github.com/mikaoelitiana/cordova-plugin-ios-no-export-compliance.git" />
-  <plugins></plugins>
+<?xml version='1.0' encoding='utf-8'?>
+<widget android-versionCode="0" id="hk.goodcity.adminstaging" ios-CFBundleVersion="0" version="0.14.017" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0">
+    <name>S. Admin GoodCity</name>
```
Comment (Darshan-Chauhan) [inline cordova/config.xml]: Can we use 2 spaces convention here over 4 space as we've used 2 space convention everywhere @shivanibhanwal ?

File: cordova/build.json [hunk 2]
```diff
@@ -1,12 +1,15 @@
 {
   "ios": {
     "debug": {
-      "codeSignIdentity": "iOS Distribution",
-      "provisioningProfile": "65f6a621-7665-44e2-b686-93fd24bf4cd5"
+      "codeSignIdentity": "iPhone Developer",
+      "developmentTeam": "6B8FS8W94M",
+      "automaticProvisioning": true
```
Comment (Darshan-Chauhan) [inline cordova/build.json]: This won't work as we've 2 provisioning profile on CI, one for staging and one for prod so XCode gets confused between two profiles so we dont need this

File: Rakefile [hunk 3]
```diff
@@ -139,8 +139,11 @@ namespace :cordova do
   desc "Cordova build {platform}"
   task build: :prepare do
     Dir.chdir(CORDOVA_PATH) do
-      build = (environment == "staging" && platform == 'android') ? "debug" : "release"
-      system({"ENVIRONMENT" => environment}, "cordova compile #{platform} --#{build} --device")
+      team_id = ENV['IOS_DEVELOPMENT_TEAM_ID']
+      provisioning_profile = ENV['PROVISIONING_PROFILE_STAGING']
+      provisioning_profile = ENV['PROVISIONING_PROFILE_PROD'] if(environment === 'production')
```
Comment (Darshan-Chauhan) [inline Rakefile]: @shivanibhanwal  You'll need to add these variables to CI env.
For that you'll need to create provisioning profile for staging and prod using development certificate and replace current pro. profile with these new ones. And use hash of that provisioning profile to CI env variables.

Comment (shivanibhanwal) [reply Rakefile]: I have already added these ENV variables in CI for the staging. Then only build has passed on circleCI. Also I have uploaded the certificate too. But I did not add it for Prod as I am still concerned will iOS support Developer license for PROD.

File: cordova/config.xml (hunk 1 above)
Comment (shivanibhanwal) [reply cordova/config.xml]: This I can fix it is my new editor I did not paid attention sorry about that.

**Pull request merged**